<!DOCTYPE html>
<html lang="en-us">
<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="theme" content="hugo-academic">
  <meta name="generator" content="Hugo 0.44" />
  <meta name="author" content="Derek Sollberger">

  
  
  
  
    
      
    
  
  <meta name="description" content="“Bioconductor provides tools for the analysis and comprehension of high-throughput genomic data. Bioconductor uses the R statistical programming language, and is open source and open development.”
https://www.bioconductor.org/
library(&quot;dplyr&quot;)library(&quot;ggplot2&quot;)InstallationTo install core packages, type the following in an R command window.
This may take around 5 minutesWhen the option for updating packages appears, type in “a” for “all”#leave as eval = FALSE when knittingif (!">

  
  <link rel="alternate" hreflang="en-us" href="/post/introduction-to-bioconductor/">

  


  

  
  
  <meta name="theme-color" content="#3f51b5">
  
  
  
  
    
  
  
    
    
      
        <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
      
    
  
  
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha512-6MXa8B6uaO18Hid6blRMetEIoPqHf7Ux1tnyIQdpt9qI5OACx7C+O3IVTr98vwGnlcg0LOLa02i9Y1HpVhlfiw==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.1/css/academicons.min.css" integrity="sha512-NThgw3XKQ1absAahW6to7Ey42uycrVvfNfyjqcFNgCmOCQ5AR4AO0SiXrN+8ZtYeappp56lk1WtvjVmEa+VR6A==" crossorigin="anonymous">
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha512-SfTiTlX6kk+qitfevl/7LibUOeJWlt9rbyDn92a1DqWOw9vWG2MFoays0sgObmWazO5BQPiFucnnEAjpAB+/Sw==" crossorigin="anonymous">
  
  
  
  
  <link rel="stylesheet" href="//fonts.googleapis.com/css?family=Montserrat:400,700%7cRoboto:400,400italic,700%7cRoboto&#43;Mono">
  
  <link rel="stylesheet" href="/styles.css">
  

  

  
  <link rel="alternate" href="/index.xml" type="application/rss+xml" title="The Median Data Scientist">
  <link rel="feed" href="/index.xml" type="application/rss+xml" title="The Median Data Scientist">
  

  <link rel="manifest" href="/site.webmanifest">
  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/icon-192.png">

  <link rel="canonical" href="/post/introduction-to-bioconductor/">

  <meta property="twitter:card" content="summary_large_image">
  
  <meta property="twitter:site" content="@DerekSollberger">
  <meta property="twitter:creator" content="@DerekSollberger">
  
  <meta property="og:site_name" content="The Median Data Scientist">
  <meta property="og:url" content="/post/introduction-to-bioconductor/">
  <meta property="og:title" content="Introduction to Bioconductor | The Median Data Scientist">
  <meta property="og:description" content="“Bioconductor provides tools for the analysis and comprehension of high-throughput genomic data. Bioconductor uses the R statistical programming language, and is open source and open development.”
https://www.bioconductor.org/
library(&quot;dplyr&quot;)library(&quot;ggplot2&quot;)InstallationTo install core packages, type the following in an R command window.
This may take around 5 minutesWhen the option for updating packages appears, type in “a” for “all”#leave as eval = FALSE when knittingif (!">
  <meta property="og:locale" content="en-us">
  
  <meta property="article:published_time" content="2020-02-12T00:00:00&#43;00:00">
  
  <meta property="article:modified_time" content="2020-02-12T00:00:00&#43;00:00">
  

  

  <title>Introduction to Bioconductor | The Median Data Scientist</title>

</head>
<body id="top" data-spy="scroll" data-target="#toc" data-offset="71" >

<nav class="navbar navbar-default navbar-fixed-top" id="navbar-main">
  <div class="container">

    
    <div class="navbar-header">
      
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse"
              data-target=".navbar-collapse" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      
      <a class="navbar-brand" href="/">The Median Data Scientist</a>
    </div>

    
    <div class="collapse navbar-collapse">

      
      
      <ul class="nav navbar-nav navbar-right">
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#about">
            
            <span>Home</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#posts">
            
            <span>Posts</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#teaching">
            
            <span>Teaching</span>
            
          </a>
        </li>

        
        

        
        
        
        
        
          
        

        <li class="nav-item">
          <a href="/#contact">
            
            <span>Contact</span>
            
          </a>
        </li>

        
        
      

      
      </ul>

    </div>
  </div>
</nav>


<article class="article" itemscope itemtype="http://schema.org/Article">

  


  <div class="article-container">
    <div class="article-inner">
      <h1 itemprop="name">Introduction to Bioconductor</h1>

      

<div class="article-metadata">

  <span class="article-date">
    
    <time datetime="2020-02-12 00:00:00 &#43;0000 UTC" itemprop="datePublished dateModified">
      Feb 12, 2020
    </time>
  </span>
  <span itemscope itemprop="author publisher" itemtype="http://schema.org/Person">
    <meta itemprop="name" content="Derek Sollberger">
  </span>

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    4 min read
  </span>
  

  
  

  
  
  
  <span class="middot-divider"></span>
  <span class="article-categories">
    <i class="fa fa-folder"></i>
    
    <a href="/categories/bioinformatics">bioinformatics</a
    >
    
  </span>
  
  

  
  
<div class="share-box" aria-hidden="true">
  <ul class="share">
    <li>
      <a class="twitter"
         href="https://twitter.com/intent/tweet?text=Introduction%20to%20Bioconductor&amp;url=%2fpost%2fintroduction-to-bioconductor%2f"
         target="_blank" rel="noopener">
        <i class="fa fa-twitter"></i>
      </a>
    </li>
    <li>
      <a class="facebook"
         href="https://www.facebook.com/sharer.php?u=%2fpost%2fintroduction-to-bioconductor%2f"
         target="_blank" rel="noopener">
        <i class="fa fa-facebook"></i>
      </a>
    </li>
    <li>
      <a class="linkedin"
         href="https://www.linkedin.com/shareArticle?mini=true&amp;url=%2fpost%2fintroduction-to-bioconductor%2f&amp;title=Introduction%20to%20Bioconductor"
         target="_blank" rel="noopener">
        <i class="fa fa-linkedin"></i>
      </a>
    </li>
    <li>
      <a class="weibo"
         href="http://service.weibo.com/share/share.php?url=%2fpost%2fintroduction-to-bioconductor%2f&amp;title=Introduction%20to%20Bioconductor"
         target="_blank" rel="noopener">
        <i class="fa fa-weibo"></i>
      </a>
    </li>
    <li>
      <a class="email"
         href="mailto:?subject=Introduction%20to%20Bioconductor&amp;body=%2fpost%2fintroduction-to-bioconductor%2f">
        <i class="fa fa-envelope"></i>
      </a>
    </li>
  </ul>
</div>


  

</div>


      <div class="article-style" itemprop="articleBody">
        <p>“<a href="https://www.bioconductor.org/">Bioconductor</a> provides tools for the analysis and comprehension of high-throughput genomic data. Bioconductor uses the R statistical programming language, and is open source and open development.”</p>
<p><a href="https://www.bioconductor.org/">https://www.bioconductor.org/</a></p>
<pre class="r"><code>library(&quot;dplyr&quot;)
library(&quot;ggplot2&quot;)</code></pre>
<div id="installation" class="section level1">
<h1>Installation</h1>
<p>To install core packages, type the following in an R command window.</p>
<ul>
<li>This may take around 5 minutes</li>
<li>When the option for updating packages appears, type in “a” for “all”</li>
</ul>
<pre class="r"><code>#leave as eval = FALSE when knitting

if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)
BiocManager::install()</code></pre>
<div id="installing-bioconductor-packages" class="section level2">
<h2>Installing Bioconductor packages</h2>
<p>There are about 20000 packages in the <em>Bioconductor</em> universe. For now, let us install a couple of the more popular packages with the following code. To install a <code>BioConductor</code> package, simply type in the name of the package in quotes inside of <code>BiocManager::install()</code></p>
<ul>
<li>This may take around 5 minutes</li>
<li>Hereafter, when the option for updating packages appears, type in “n” for “no”</li>
</ul>
<pre class="r"><code>#leave as eval = FALSE when knitting

BiocManager::install(&quot;GenomicFeatures&quot;)
BiocManager::install(&quot;Biostrings&quot;)</code></pre>
</div>
</div>
<div id="exploring-genomicfeatures" class="section level1">
<h1>Exploring GenomicFeatures</h1>
<p>The following instructions were extracted from the vignette for Genomic Features</p>
<ul>
<li><code>browseVignettes(package = "GenomicFeatures")</code></li>
</ul>
<p>Load the <code>GenomicFeatures</code> package. It is suggested that you load the packages with the following code.</p>
<pre class="r"><code>#set as eval = TRUE when knitting

suppressPackageStartupMessages(library(&#39;GenomicFeatures&#39;))</code></pre>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>“The <code>GenomicFeatures</code> package retrieves and manages transcript-related features fromthe UCSC Genome Bioinformatics and BioMart data resources. The package isuseful for ChIP-chip, ChIP-seq, and RNA-seq analyses.”</p>
</div>
<div id="txdb-objects" class="section level2">
<h2>TxDb Objects</h2>
<p>"The <code>GenomicFeatures</code> package uses TxDb objects to store transcript metadata. This class maps the 5’ and 3’ untranslated regions (UTRs), protein coding sequences(CDSs) and exons for a set of mRNA transcripts to their associated genome. TxDb objects have numerous accessors functions to allow such features to be retrieved individually or grouped together in a way that reflects the underlying biology.</p>
<p>All TxDb objects are backed by a SQLite database that manages genomic locations and the relationships between pre-processed mRNA transcripts, exons, protein coding sequences, and their related gene identifiers."</p>
<pre class="r"><code>#set as eval = TRUE when knitting

samplefile &lt;- system.file(&quot;extdata&quot;,
                          &quot;hg19_knownGene_sample.sqlite&quot;,
                          package=&quot;GenomicFeatures&quot;)
txdb &lt;- loadDb(samplefile)
txdb</code></pre>
<ol style="list-style-type: decimal">
<li>Write a couple of sentences about what you observe in the output (the description of the txdb sample file)</li>
</ol>
</div>
<div id="loading-and-viewing-a-sequence" class="section level2">
<h2>Loading and Viewing a Sequence</h2>
<ol start="2" style="list-style-type: decimal">
<li>Install the following packages (i.e. figure out how from some code earlier in this tutorial)</li>
</ol>
<ul>
<li><code>BSgenome.Hsapiens.UCSC.hg19</code></li>
<li><code>TxDb.Hsapiens.UCSC.hg19.knownGene</code></li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li>Run the following code and then write a couple of sentences about what you observe in the output.</li>
</ol>
<pre class="r"><code>#set as eval = TRUE when knitting

library(BSgenome.Hsapiens.UCSC.hg19)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)

tx_seqs1 &lt;- extractTranscriptSeqs(Hsapiens,
                                  TxDb.Hsapiens.UCSC.hg19.knownGene,
                                  use.names=TRUE)
suppressWarnings(translate(tx_seqs1))</code></pre>
</div>
</div>
<div id="exploring-biostrings" class="section level1">
<h1>Exploring Biostrings</h1>
<p>The following instructions were extracted from the vignette for <code>Biostrings</code></p>
<ul>
<li><code>browseVignettes(package = "Biostrings")</code></li>
</ul>
<ol start="4" style="list-style-type: decimal">
<li>Install the following packages (i.e. figure out how from some code earlier in this tutorial)</li>
</ol>
<ul>
<li><code>hgu95av2probe</code></li>
<li><code>hgu95av2cdf</code></li>
</ul>
<p>Load the <code>Biostrings</code> package. It is suggested that you load the packages with the following code.</p>
<pre class="r"><code>#set as eval = TRUE when knitting

suppressPackageStartupMessages(library(&#39;Biostrings&#39;))
suppressPackageStartupMessages(library(&#39;hgu95av2probe&#39;))
suppressPackageStartupMessages(library(&#39;hgu95av2cdf&#39;))</code></pre>
<ol start="5" style="list-style-type: decimal">
<li>The following code will load an example of a “Large DNAStringSet” and perform a couple of simple data analyses. Run the code and write a couple of sentences about the results.</li>
</ol>
<pre class="r"><code>#set as eval = TRUE when knitting

this_DNAString_set &lt;- DNAStringSet(hgu95av2probe)
this_DNAString_set_bc &lt;- alphabetFrequency(this_DNAString_set, baseOnly = TRUE)
nrow(this_DNAString_set_bc)
head(this_DNAString_set_bc)
alphabetFrequency(this_DNAString_set_bc, baseOnly=TRUE, collapse=TRUE)</code></pre>
<ol start="6" style="list-style-type: decimal">
<li>The following code will perform a calculation called the <code>GC content</code> of a DNA sequence and graph the resulting amounts as a histogram. Run the code and write a couple of sentences about the results.</li>
</ol>
<pre class="r"><code>#set as eval = TRUE when knitting

this_DNAString_df &lt;- data.frame(this_DNAString_set_bc)

# alas &quot;T&quot; is treated as &quot;TRUE&quot; in R, so let&#39;s use the full names of the nucleotides
colnames(this_DNAString_df) &lt;- c(&quot;Adenine&quot;, &quot;Cytosine&quot;, &quot;Guanine&quot;, &quot;Thymine&quot;, &quot;other&quot;)

this_DNAString_df %&gt;%
  select(c(&quot;Adenine&quot;, &quot;Cytosine&quot;, &quot;Guanine&quot;, &quot;Thymine&quot;)) %&gt;%
  mutate(GC_content = 100*(Cytosine + Guanine) / 
           (Adenine + Cytosine + Guanine + Thymine)) %&gt;%
  ggplot(aes(x = GC_content, fill = ..x..)) +
  geom_histogram(binwidth = 5) + 
  labs(title = &quot;Affymetrix Human Genome U95 Set annotation data&quot;,
       subtitle = &quot;GC Content of hgu95av2probe&quot;,
       caption = &quot;Source: Bioconductor&quot;,
       x = &quot;GC content (percentage)&quot;) +
  scale_fill_gradient(low = &quot;yellow&quot;, high= &quot;blue&quot;) +
  theme_minimal() +
  theme(legend.position = &quot;none&quot;)</code></pre>
</div>

      </div>

      




    </div>
  </div>

</article>






<div class="article-container">
  

</div>

<footer class="site-footer">
  <div class="container">
    <p class="powered-by">

      &copy; 2018 &middot; 

      Powered by the
      <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
      <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

      <span class="pull-right" aria-hidden="true">
        <a href="#" id="back_to_top">
          <span class="button_icon">
            <i class="fa fa-chevron-up fa-2x"></i>
          </span>
        </a>
      </span>

    </p>
  </div>
</footer>


<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <button type="button" class="close btn-large" data-dismiss="modal">&times;</button>
        <h4 class="modal-title">Cite</h4>
      </div>
      <div>
        <pre><code class="modal-body tex"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-primary btn-outline js-copy-cite" href="#" target="_blank">
          <i class="fa fa-copy"></i> Copy
        </a>
        <a class="btn btn-primary btn-outline js-download-cite" href="#" target="_blank">
          <i class="fa fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

    

    
    

    

    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.min.js" integrity="sha512-3P8rXCuGJdNZOnUx/03c1jOTnMn3rP63nBip5gOP2qmUh5YAdVAvFZ1E+QLZZbC1rtMrQb+mah3AfYW11RUrWA==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.3/imagesloaded.pkgd.min.js" integrity="sha512-umsR78NN0D23AzgoZ11K7raBD+R6hqKojyBZs1w8WvYlsI+QuKRGBx3LFCwhatzBunCjDuJpDHwxD13sLMbpRA==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha512-iztkobsvnjKfAtTNdHkGVjAYTrrtlC7mGp/54c40wowO7LhURYl3gVzzcEqGl/qKXQltJ2HwMrdLcNUdo+N/RQ==" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.4/isotope.pkgd.min.js" integrity="sha512-VDBOIlDbuC4VWxGJNmuFRQ0Li0SKkDpmGyuhAG5LTDLd/dJ/S0WMVxriR2Y+CyPL5gzjpN4f/6iqWVBJlht0tQ==" crossorigin="anonymous"></script>
    
    
    <script src="/js/hugo-academic.js"></script>
    

    
    
      
      
      <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js" integrity="sha256-/BfiIkHlHoVihZdc6TFuj7MmJ0TWcWsMXkeDFwhi0zw=" crossorigin="anonymous"></script>
      

      

      

      <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script type="text/x-mathjax-config">
        MathJax.Hub.Config({ tex2jax: { inlineMath: [['$','$'], ['\\(','\\)']] } });
    </script>
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS_CHTML" integrity="sha512-tOav5w1OjvsSJzePRtt2uQPFwBoHt1VZcUq8l8nm5284LEKE9FSJBQryzMBzHxY5P0zRdNqEcpLIRVYFNgu1jw==" crossorigin="anonymous"></script>
    
    

  </body>
</html>

